lib/Core/SpecialFunctionHandler.cpp

```diff
@@ -427,6 +428,30 @@ void SpecialFunctionHandler::handleMalloc(ExecutionState &state,
   executor.executeAlloc(state, arguments[0], false, target);
 }
 
+void SpecialFunctionHandler::handleMemalign(ExecutionState &state,
+                                            KInstruction *target,
+                                            std::vector<ref<Expr>> &arguments) {
+  assert(arguments.size() == 2 && "invalid number of arguments to memalign");
+
+  std::pair<ref<Expr>, ref<Expr>> alignmentRangeExpr =
+      executor.solver->getRange(state, arguments[0]);
+  ref<Expr> alignmentExpr = alignmentRangeExpr.first;
+  auto alignmentConstExpr = dyn_cast<ConstantExpr>(alignmentExpr);
+  assert(alignmentConstExpr &&
```

Comment:
Be graceful here as well.

Reply:
The Solver::getRange function never returns anything else than a pair of ref<ConstantExpr>. I wrote the assert to show that the dyn_cast is expected to never fail.

Reply:
The current interface description says:
```
    /// \return - A pair with (min, max) values for the expression.
    ///
    /// \post(mustBeTrue(min <= e <= max) && 
    ///       mayBeTrue(min == e) &&
    ///       mayBeTrue(max == e))
```
It says nothing that expressions have to be `ConstantExpr`.
Yes, it currently does return a pair of `ConstantExpr` but that is implementation and can change.

Reply:
Okay, I changed the assert to a terminateState